diff -uNr a/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp b/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp	2017-11-14 08:55:01.000000000 -0800
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp	2018-05-20 20:56:35.541040026 -0700
@@ -408,7 +408,9 @@
   {
     case AV_CODEC_ID_MPEG1VIDEO:
     case AV_CODEC_ID_MPEG2VIDEO:
+#if FF_API_XVMC
     case AV_CODEC_ID_MPEG2VIDEO_XVMC:
+#endif
       format = VFORMAT_MPEG12;
       break;
     case AV_CODEC_ID_H263:
@@ -1161,7 +1163,10 @@
         }
         if (( AV_CODEC_ID_MPEG1VIDEO == para->video_codec_id)
           || (AV_CODEC_ID_MPEG2VIDEO == para->video_codec_id)
-          || (AV_CODEC_ID_MPEG2VIDEO_XVMC == para->video_codec_id)) {
+#if FF_API_XVMC
+          || (AV_CODEC_ID_MPEG2VIDEO_XVMC == para->video_codec_id)
+#endif
+          ) {
             ret = mpeg_add_header(para, pkt);
             if (ret != PLAYER_SUCCESS) {
                 return ret;
diff -uNr a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp	2017-11-14 08:55:01.000000000 -0800
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp	2018-05-20 20:55:32.591145510 -0700
@@ -85,7 +85,9 @@
       break;
     case AV_CODEC_ID_MPEG1VIDEO:
     case AV_CODEC_ID_MPEG2VIDEO:
+#if FF_API_XVMC
     case AV_CODEC_ID_MPEG2VIDEO_XVMC:
+#endif
       if (m_hints.width <= CSettings::GetInstance().GetInt(CSettings::SETTING_VIDEOPLAYER_USEAMCODECMPEG2))
         return false;
       m_mpeg2_sequence_pts = 0;
diff -uNr a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp	2017-11-14 08:55:01.000000000 -0800
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp	2018-05-20 20:56:59.031001734 -0700
@@ -553,7 +553,9 @@
     m_pFormatName = "iMX-mpeg1";
     break;
   case AV_CODEC_ID_MPEG2VIDEO:
+#if FF_API_XVMC
   case AV_CODEC_ID_MPEG2VIDEO_XVMC:
+#endif
     m_decOpenParam.CodecFormat = VPU_V_MPEG2;
     m_pFormatName = "iMX-mpeg2";
     break;
