From cda427fba363fc9c56d4bcbe322e1ed8a6fe9a95 Mon Sep 17 00:00:00 2001
From: Mauro Ribeiro <mauro.ribeiro@hardkernel.com>
Date: Tue, 6 Jan 2015 02:19:20 +1030
Subject: [PATCH] [aml] add ODROID as a compatible for Meson8B

---
 xbmc/utils/AMLUtils.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/utils/AMLUtils.cpp b/xbmc/utils/AMLUtils.cpp
index c127c84..8fa3d15 100644
--- a/xbmc/utils/AMLUtils.cpp
+++ b/xbmc/utils/AMLUtils.cpp
@@ -136,7 +136,7 @@ enum AML_DEVICE_TYPE aml_get_device_type()
         aml_device_type = AML_DEVICE_TYPE_M8M2;
       else
         aml_device_type = AML_DEVICE_TYPE_M8;
-    } else if (cpu_hardware.find("Meson8B") != std::string::npos)
+    } else if ((cpu_hardware.find("Meson8B") != std::string::npos) || (cpu_hardware.find("ODROID") != std::string::npos))
       aml_device_type = AML_DEVICE_TYPE_M8B;
     else
       aml_device_type = AML_DEVICE_TYPE_UNKNOWN;
